AbstractDesignPatterns/Program.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+
+namespace AbstractDesignPatterns
+{
+
+    class Program
+    {
+        static void Main(string[] args) // client 
+        {
+
+            IPhoneProduct appleFactory = new AppleFactory();
```

Comment:
You have declared `appleFactory` using `IPhoneProduct` it is like having a factory of type product! instead it should be like this 
```c#
AbstractFactory appleFactory = new AppleFactory();
```